### PR TITLE
TableReporter: Apply some minor code simplifications

### PR DESCRIPTION
**Pull request by sschuberth**
This change is [Reviewable](https://reviewable.io/reviews/heremaps/oss-review-toolkit/945)

### CODE REVIEW

File: analyzer/src/main/kotlin/managers/Maven.kt [hunk 1]
```diff
@@ -66,13 +66,10 @@ class Maven(analyzerConfig: AnalyzerConfiguration, repoConfig: RepositoryConfigu
...
-            return if (artifact.extension == "pom") {
-                localProjectBuildingResults[artifact.identifier()]?.pomFile?.absoluteFile
-            } else {
-                null
-            }
-        }
+        override fun findArtifact(artifact: Artifact) =
+                localProjectBuildingResults[artifact.identifier()]?.pomFile?.absoluteFile.takeIf {
+                    artifact.extension == "pom"
+                }
```
Comment (mnonnenmacher) [inline analyzer/src/main/kotlin/managers/Maven.kt]: Using `takeIf` without using `it` inside seems like a misuse of the function. Also, it moves the check here, so there will be unnecessary map lookups. What about using a helper function for such cases:

```kotlin
override fun findArtifact(artifact: Artifact) =
    ifOrNull(artifact.extension == "pom") {
        localProjectBuildingResults[artifact.identifier()]?.pomFile?.absoluteFile
    }

inline fun <T> ifOrNull(condition: Boolean, block: () -> T) = if (condition) block() else null
```

Comment (sschuberth) [reply analyzer/src/main/kotlin/managers/Maven.kt]: Good point about the potentially unnecessary map lookup. I've reverted the `takeIf` changes as I don't like to introduce yet another helper function just for this case.

Comment (mnonnenmacher) [reply analyzer/src/main/kotlin/managers/Maven.kt]: I just realized there is a way without another helper by moving the `takeIf`:

```kotlin
override fun findArtifact(artifact: Artifact) =
    artifact.takeIf { it.extension == "pom" }?.let {
        localProjectBuildingResults[it.identifier()]?.pomFile?.absoluteFile
    }
```

I'll approve and leave it up to you if you want to change it again.

Comment (sschuberth) [reply analyzer/src/main/kotlin/managers/Maven.kt]: Good idea, I'll add a new commit.

**Pull request merged**